src/libxtp/gauss_hermite_quadrature_constants.cc

```diff
@@ -0,0 +1,367 @@
+/*
+ *            Copyright 2009-2019 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSe-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <votca/xtp/eigen.h>
+#include <votca/xtp/gauss_hermite_quadrature_constants.h>
+
+using namespace std;
+
+namespace votca {
+namespace xtp {
+
+const Eigen::VectorXd& Gauss_Hermite_Quadrature_Constants::getPoints(
+    Index order) {
+  if (!this->_filled_Points) {
+    this->FillPoints();
+    _filled_Points = true;
+  }
+  if (_map_points.count(order) == 0)
+    throw invalid_argument("Quadrature order " + std::to_string(order) +
+                           " not in {8,10,12,14,16,18,20 and 100}. Please "
+                           " select one of these numbers");
+  return _map_points.at(order);
+}
```

Comment:
Also here I think I am not following you :D

Reply:
So the `getPoints()` function cannot be called inside an `openmp` loop, because you could get race conditions there. So I think we should do the map stuff differently. I have to think about it. My suggestion was to put the `FillPoints()` into the constructor of this class and delete the `if(_filled_Points)` clause and the stuff inside.